File: trace_dom.py
import sys

def trace_html(file_path):
    with open(file_path, 'r', encoding='utf-8') as f:
        lines = f.readlines()
        
    depth = 0
    in_app_container = False
    
    for i, line in enumerate(lines):
        line_num = i + 1
        
        # Start tracking from app-container
        if 'class="app-container"' in line:
            in_app_container = True
            print(f"[{line_num}] START APP-CONTAINER")
            
        if not in_app_container:
            continue
            
        # Count divs
        opens = line.count('<div')
        closes = line.count('</div')
        
        for _ in range(opens):
            depth += 1
            if 'id="salezhub-tab"' in line or 'id="ceoContent"' in line or 'id="synchub-tab"' in line:
                print(f"[{line_num}] {'  ' * depth} OPEN (Depth: {depth}): {line.strip()}")
                
        for _ in range(closes):
            if depth == 1:
                print(f"[{line_num}] APP-CONTAINER CLOSED AT LINE {line_num}!")
                return
            depth -= 1

trace_html('index.html')

File: trace_dom.js
const fs = require('fs');
const lines = fs.readFileSync('index.html', 'utf-8').split('\n');
let depth = 0;
let inApp = false;

for (let i = 0; i < lines.length; i++) {
    const line = lines[i];
    if (line.includes('class="app-container"')) {
        inApp = true;
        console.log(`[${i+1}] START APP-CONTAINER`);
    }
    if (!inApp) continue;
    
    // rough count
    let opens = (line.match(/<div/g) || []).length;
    let closes = (line.match(/<\/div/g) || []).length;
    
    for (let j=0; j < opens; j++) {
        depth++;
        if (line.includes('id="salezhub-tab"') || line.includes('id="synchub-tab"') || line.includes('ceoContent') || line.includes('ceo-tab-inner')) {
            console.log(`[${i+1}] ${'  '.repeat(depth)} OPEN (Depth: ${depth}): ${line.trim()}`);
        }
    }
    
    for (let j=0; j < closes; j++) {
        if (depth === 1) {
            console.log(`[${i+1}] APP-CONTAINER CLOSED AT LINE ${i+1}!`);
            process.exit(0);
        }
        depth--;
        if (line.includes('<!-- End of sales-tab -->')) {
            console.log(`[${i+1}] ${'  '.repeat(depth)} CLOSED salezhub-tab (Depth: ${depth})`);
        }
    }
}
